build.gradle

```diff
@@ -196,6 +196,31 @@ subprojects {
   }
 }
 
+task collectTestXml (group: 'Test reports') {
+  description = "Collects the test xmls from every subproject into a new directory with subproject as name."
+  def reportingProjects = subprojects.findAll()
+  dependsOn(reportingProjects.tasks.collectMany{it.withType(Test)})
+
+
+  doFirst {
+    def target = file("$buildDir/collected-test-results")
+    target.mkdirs()
+    reportingProjects.each { rp ->
+      rp.tasks.withType(Test).each { testTask ->
+        def repSrc = testTask.reports.junitXml.destination
+
+        def repTarget = file("$target/$rp.name-$repSrc.name")
+        copy {
+          from repSrc
+          into repTarget
+          include '**/*.xml'
```

Comment:
@marcphilipp Doesn't Gradle have any easier way to have aggregated test results?

Reply:
@leonard84 Why is this necessary? Does CircleCI not support wildcards?

Reply:
Yup they don't https://circleci.com/docs/2.0/configuration-reference/#store_test_results